### DISCUSSION TITLE
Tasmota ESP8266 220V board

### DISCUSSION THREAD

**Discussion by lionlggnd** [arendst/Tasmota#18695] [Q&A]
i wanna try to build a board that picks the 220V and convert it to 3,3V to run a ESP8266 on the board.
i'm making pcb and schematics on easyeda and i want to review it with you expecially about the RST and gpio0 (button) because i dont know if i made well the connections, because i'm still a newbie in this argument.
thanks
any report or suggestion appreciated !!
![image](https://github.com/arendst/Tasmota/assets/117315249/cced43ba-8768-468c-8274-c7808413ec75)
![image](https://github.com/arendst/Tasmota/assets/117315249/5d8a2e7a-fab9-46ec-946b-3987cf47eaf1)

**Comment by barbudor** [comment]
The schematic is ok
The routing is extremely poor and dangerous

DO NOT HAVE THE LOW VOLTAGE TRACK CLOSE TO THE AC CONNECTION

**Comment by lalo-uy** [comment]
Why all so distant.?
If you rotate the buttons 90deg the wiring is mutch easy.
Use a poligon for ground.

**Comment by Wintespe** [comment]
Unfortunately we don't know what it looks like inside the HLK-PM03. Therefore please insert a fuse e.g. T1.0A in the 230V line.

**Comment by lionlggnd** [comment]
the pcb is a small draft. I would have placed that 3.3V line in another place also because in this way I would not have been able to add the terminal for the 220V. I would like to know beyond that the modifications I have to make, if according to this scheme the chip will work, and if the connections of the buttons to gpio0 and RST are done correctly. i have seen from somewhere that i have to bring 3.3V to the RST pin for it to work too. It is true?

**Comment by lalo-uy** [comment]
You need pull up resistors on RST and GPIO0.
100nF cap between vcc and ground just beside the vcc pin is good.

**Comment by lionlggnd** [comment]
>>The schematic is ok The routing is extremely poor and dangerous

>>DO NOT HAVE THE LOW VOLTAGE TRACK CLOSE TO THE AC CONNECTION

Low voltage track is now ok.
What i else should do?

**Comment by JaffaSweet** [comment]
wow very interesting

**Comment by lionlggnd** [comment]
nobody can help me? I would like to know especially if the positioning of the resistors is right and if the chip will work, and if I have to make further additions. Thank you

**Comment by Jason2866** [comment]
This is Tasmota Discussions. It is NOT the place where PCB Hardware designs are discussed. It is Off Topic.
Since 230V AC is involved a PCB board needs to be designed safe.
